wandb/sdk/lib/sock_client.py

```diff
@@ -64,6 +64,27 @@ def send_server_response(self, msg: Any) -> None:
             # things like network status poll loop, there might be a better way to quiesce
             pass
 
+    def send_and_recv(
+        self,
+        *,
+        inform_init: spb.ServerInformInitRequest = None,
+        inform_start: spb.ServerInformStartRequest = None,
+        inform_attach: spb.ServerInformAttachRequest = None,
+        inform_finish: spb.ServerInformFinishRequest = None,
+        inform_teardown: spb.ServerInformTeardownRequest = None
+    ) -> spb.ServerResponse:
+        self.send(
+            inform_init=inform_init,
+            inform_start=inform_start,
+            inform_attach=inform_attach,
+            inform_finish=inform_finish,
+            inform_teardown=inform_teardown,
+        )
+        response = self.read_server_response(timeout=1)
+        if response is None:
```

Comment:
When could this be happening? Might be nice to have a test case for this.

Reply:
Also, what happens downstream if it is None?

Reply:
hmmm... it is a good question... this is the relevant function:
https://github.com/wandb/client/blob/master/wandb/sdk/lib/sock_client.py#L118
so it seems the `None` is a failure state... I don't know how to add tests here. Could it be part of the error injection? @raubitsj

Reply:
now I raise an exception, but we could send a None and just create an empty message or an error message... was thinking of doing it as part of the error handling PR: https://github.com/wandb/client/pull/3336